lib/lru.d.ts

```diff
@@ -0,0 +1,18 @@
+declare class LRU<T = any> {
+    private first;
+    private items;
+    private last;
+    private max;
+    private size;
+    private ttl;
+    constructor(max?: number, ttl?: number);
+    has(key: string): boolean;
+    clear(): this;
+    delete(key: string): this;
+    evict(): this;
+    get(key: string): T;
+    keys(): string[];
+    set(key: string, value: T, bypass?: boolean): this;
+}
+export default function factory(max?: number, ttl?: number): LRU<any>;
+export {};

```

Comment:
The auto generated Type Definitions.

Reply:
There is a problem with a missing generics in factory. We need to expose `factory<T = any>(...)` to allow types being handled and propagated to the rest of the user code.

Reply:
yeah let me look into this more. TS compiler seems to think otherwise.

Reply:
What do you mean with "TS compiler seems to think otherwise"?